obs2ioda-v2/src/f_c_string_t_mod.f90

```diff
@@ -0,0 +1,128 @@
+! Module providing functionality for handling a single Fortran string as a
+! C-compatible null-terminated string.
+module f_c_string_t_mod
+    use iso_c_binding, only : c_loc, c_ptr, c_null_char, &
+            c_char, c_f_pointer, c_null_ptr, c_size_t, c_associated, c_int, c_size_t
+    implicit none
+
+    public :: strlen
+
+    public :: f_c_string_t
+
+    interface
+        ! Declares an interface for the C function `strlen` to calculate the length
+        ! of a null-terminated C string.
+        !
+        ! Arguments:
+        ! - c_string: A C pointer to the null-terminated string whose length is to be calculated.
+        !
+        ! Returns:
+        ! - n: The length of the C string as an integer of kind `c_size_t`.
+        function c_strlen(c_string) bind(C, name = "strlen") result(n)
```

Comment:
In my opinion, it's not good practice to bind the name of a Fortran interface to a different name in C. Probably better to let `strlen` refer to the C `strlen` function, and to choose a different name for any wrapper around `strlen` that you choose to define elsewhere in this module.